language-support/scala/codegen-sample-app/src/test/scala/com/digitalasset/codegen/ScalaCodeGenIT.scala

```diff
@@ -98,6 +98,14 @@ class ScalaCodeGenIT
   override protected def beforeAll(): Unit = {
     super.beforeAll()
     ledger = Await.result(LedgerClient(channel, clientConfig), StartupTimeout)
+    val _ = Await.ready(
+      Future.sequence(
+        List(alice, bob, charlie).map(p =>
+          ledger.partyManagementClient.allocateParty(Some(p.toString), None)
+        )
+      ),
+      StartupTimeout,
+    )
```

Comment:
can we have here a single `Await` on the `LedgerClient` creation as well as party allocation?

And independently of that - is allocateParty idempotent? Can it be part of the each relevant test?

Reply:
:+1:  for the first point
No, `allocateParty` sadly is not idempotent and calling it for a second time in a second test will fail. The way around it would be to generate unique parties in each test, but that would be beyond the scope of this PR.